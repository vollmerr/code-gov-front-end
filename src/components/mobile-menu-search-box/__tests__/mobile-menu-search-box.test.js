import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux'

import MobileMenuSearchBox from 'components/mobile-menu-search-box'

import { shallow, mount } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <MobileMenuSearchBox /> component', () => {
  const props = {
    onSubmit: jest.fn()
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MobileMenuSearchBox store={store} {...props} />);
  });
  it('renders MobileMenuSearchBox', () => {
    wrapper = wrapper.dive();
    expect(wrapper.exists()).toEqual(true);
  });
  it('search submit', () => {
    const fakeEvent = { preventDefault: jest.fn() };
    wrapper = wrapper.dive();
    wrapper.instance().handleSubmit(fakeEvent);
    expect(wrapper.state().value).toEqual('');
  });
  it('handleChange', () => {
    wrapper = wrapper.dive();
    wrapper.instance().handleChange({target: {value: 'a'}});
    expect(wrapper.state().value).toEqual('a');
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
})