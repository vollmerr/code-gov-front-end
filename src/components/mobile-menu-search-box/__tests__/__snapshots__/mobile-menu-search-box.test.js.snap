// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MobileMenuSearchBox /> component should render correctly 1`] = `
<MobileMenuSearchBoxComponent
  onSubmit={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
